app/src/main/java/com/mapbox/navigation/examples/camera/ShowCameraTransitionsActivity.kt

```diff
@@ -65,11 +66,10 @@ import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
-    )
-
+    private val route by lazy {
+        DirectionsRoute.fromJson(getString(R.string.hardcoded_route_json))
+            .toNavigationRoute()
```

Comment:
I need some advice here. The docs for `toNavigationRoute` say it's potentially time-consuming so it's not cool to call it from the main thread. However, it was called from the main thread earlier when the deprecated API was used, for example here's the `setRoutes` implementation:
```
    fun setRoutes(routes: List<DirectionsRoute>, initialLegIndex: Int = 0) {
        setNavigationRoutes(routes.toNavigationRoutes(), initialLegIndex)
    }
```
And I don't see how I can create directly NavigationRoute with hardcoded data.
Could someone please share their opinion?

Reply:
Ideally, we'd have to move whatever code relies on this field to a coroutine and initiate the `NavigationRoute` async. That said, given that this is only a debug test activity, we can leave the setup as is to keep it simple.